packages/main/src/plugin/container-registry.ts

```diff
@@ -129,6 +129,7 @@ export class ContainerProviderRegistry {
 
   protected containerProviders: Map<string, containerDesktopAPI.ContainerProviderConnection> = new Map();
   protected internalProviders: Map<string, InternalContainerProvider> = new Map();
+  protected activeTimeouts: Map<string, NodeJS.Timeout | undefined> = new Map();
```

Comment:
what is the point of saving the NodeJS.Timeout ? We could just have a boolean Map since we only check for existence. See my comment here https://github.com/podman-desktop/podman-desktop/pull/10239/files#r1895614426

Would need to rename to `initializedConnectionAPI: Map<string, boolean>`